builtin/k8s/apply/platform.go

```diff
@@ -0,0 +1,297 @@
+	if err != nil {
+		return nil, err
+	}
+	if err := cmd.Run(); err != nil {
+		return nil, err
+	}
+	s.Done()
+
+	return deployment, nil
+}
+
+// Destroy
+func (p *Platform) Destroy(
+	ctx context.Context,
+	log hclog.Logger,
+	deployment *Deployment,
+	ui terminal.UI,
+) error {
+	sg := ui.StepGroup()
+	defer sg.Wait()
+	s := sg.Add("Executing kubectl to destroy...")
+	defer func() { s.Abort() }()
+
+	if deployment.PruneLabel == "" {
+		s.Update("No prune label on deployment. Not destroying.")
+		s.Done()
+		return nil
+	}
+
+	// Apply it
+	cmd, err := p.cmd(ctx, s, "delete", "all", "-l", deployment.PruneLabel)
+	if err != nil {
+		return err
+	}
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+
+	s.Done()
+	return nil
+}
+
+// Generation returns the generation ID.
+func (p *Platform) Generation(
+	ctx context.Context,
+) ([]byte, error) {
+	// Static generation since we will always use the `prune_label` to
+	// automatically delete unused resources.
+	return []byte("kubernetes-apply"), nil
+}
+
+func (p *Platform) cmd(
+	ctx context.Context,
+	step terminal.Step,
+	subcmd string, args ...string,
+) (*exec.Cmd, error) {
+	path, err := exec.LookPath("kubectl")
+	if err != nil {
+		return nil, status.Errorf(codes.FailedPrecondition,
+			"kubectl could not be found on the PATH")
+	}
+
+	// Build the args we'll send to kubectl, which are:
+	// <subcmd> <flags> <flags + args provided to this func>
+	realArgs := []string{subcmd}
+	if p.config.KubeconfigPath != "" {
+		realArgs = append(realArgs, "-kubeconfig="+p.config.KubeconfigPath)
+	}
+	realArgs = append(realArgs, args...)
+
+	// Build our command
+	cmd := exec.CommandContext(ctx, path, realArgs...)
+
+	// Ensure output goes to our step
+	cmd.Stdout = step.TermOutput()
+	cmd.Stderr = cmd.Stdout
+	return cmd, nil
+}
+
+// Config is the configuration structure for the Platform.
+type Config struct {
+	// The path to the job specification to load.
+	Path string `hcl:"path,attr"`
+
+	// Prune label is the label to use to destroy resources that don't match.
+	PruneLabel string `hcl:"prune_label,attr"`
+
+	// KubeconfigPath is the path to the kubeconfig file.
+	KubeconfigPath string `hcl:"kubeconfig,optional"`
+}
+
+func (p *Platform) Documentation() (*docs.Documentation, error) {
+	doc, err := docs.New(docs.FromConfig(&Config{}), docs.FromFunc(p.DeployFunc()))
+	if err != nil {
+		return nil, err
+	}
+
+	doc.Description(`
+Deploy Kubernetes resources directly from a single file or a directory of YAML
+or JSON files.
+
+This plugin lets you use any pre-existing set of Kubernetes resource files
+to deploy to Kubernetes. This plugin supports all the features of Waypoint.
+You may use Waypoint's [templating features](/docs/waypoint-hcl/functions/template)
+to template the resources with information such as the artifact from
+a previous build step, entrypoint environment variables, etc.
+
+### Requirements
+
+This plugin requires "kubectl" to be installed since this plugin works by
```

Comment:
Maybe add a note or instructions around Kubernetes config? Say users need to have the configuration present in order for `kubectl` to access it.

Reply:
Good call. Fixed!